llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -1024,29 +1022,16 @@ static void translateUnPackMask(CallInst &CI) {
 // This function sets VCNamedBarrierCount attribute to set
 // the number of named barriers required by a kernel
 static void translateNbarrierInit(CallInst &CI) {
-  auto *F = CI.getFunction();
-
+  auto F = CI.getParent()->getParent();
   auto *ArgV = CI.getArgOperand(0);
-  assert(isa<ConstantInt>(ArgV) &&
-         "integral constant expected for nbarrier count");
+  if (!isa<ConstantInt>(ArgV)) {
```

Comment:
There were 2 functions: 1 for SLMSize, 1 for NBarrierCount. I just unified the code (copy-pasted that from one to another).
I agree, this can be replaced with AssertRelease in both funcs.

Reply:
Thank you. Fixed in e39c9ef6a9abd9d2b94a84c9d32929554da3f841